ravenframework/utils/utils.py

```diff
@@ -547,6 +547,30 @@ def importFromPath(filename, printImporting = True):
     raise Exception('(            ) '+ UreturnPrintTag('UTILS') + ': '+UreturnPrintPostTag('ERROR')+ '-> importing module '+ filename + ' at '+path+os.sep+name+' failed with error '+str(ae))
   return importedModule
 
+def importOrInstall(package):
+  """
+    Method to import or install (if not found) a library named "package"
+    @ In, package, str, the package name to import
+    @ Out, pckImport, module, the imported module
+
+    NOTE: This method should be used only by developers
+          since it silently install packages
+          if not found
+  """
+  pckImport = None
+  try:
+    pckImport = __import__(package)
+  except ImportError:
+    print("(            ) "+UreturnPrintTag('UTILS')+": "+UreturnPrintPostTag('Message') + " -> Python package "+package+"not found. Trying to install it via pip!")
+    import subprocess
+    s = subprocess.getstatusoutput("python -m pip install " + package)
+    if int(s[0]) == 0:
+      print("(            ) "+ UreturnPrintTag('UTILS')+": "+ UreturnPrintPostTag('Message') + " -> Installation succeded!")
+      pckImport = __import__(package)
+    else:
+      print("(            ) " + UreturnPrintTag('UTILS')+ ": " + UreturnPrintPostTag('Message') + " -> Installation failed with error: " + s[1] )
+  return pckImport
```

Comment:
As stated, this method should be used only by developers. Can you add the package through RAVEN dependencies file? You may use optional attribute if you do not want to force to install it.

Reply:
Done. And also modified the test to check for that library (if not found, the tests should be skipped)

Reply:
@wangcj05 I don't remember. Are the "optional" libraries installed in any machine? (If not, the serpent regression tests are never run)

Reply:
We do have one unix machine to test optional, however, for your situation, the library serpentTools is not installed. It seems something is wrong with  the scripts/library_handler.py, the optional library is not added to `pip` for some reason. I would suggest you update the script to handle the pip with optional.

Reply:
I guess the problem is located in the following lines, the addOptional is set to False when using pip within conda.

![image](https://github.com/idaholab/raven/assets/7321071/6bdcba2f-b66e-4479-8633-e649c510bbcf)

Reply:
added.

Reply:
It seems that serpentTools is now installed.